app/src/main/java/com/codepath/michfeng/songswiper/activities/LikedActivity.java

```diff
@@ -73,25 +72,69 @@ protected void onCreate(Bundle savedInstanceState) {
             Glide.with(this).load(card.getCoverImagePath()).into(ivAlbum);
         }
 
-        // Add liked song, genre, and artist to their respective artist.
+        // Add liked song, genre, and artist to their respective queues stored in user.
+        // Query for LikedObjects object corresponding to user.
         ParseUser currentUser = ParseUser.getCurrentUser();
-        Queue<TrackSimplified> likedTracks = (LinkedList<TrackSimplified>) currentUser.get("likedTracks");
-        Queue<ArtistFull> likedArtists = (LinkedList<ArtistFull>) currentUser.get("likedArtists");
-        Queue<String> likedGenres = (LinkedList<String>) currentUser.get("likedGenres");
 
-        likedTracks.add(card.getTrack());
-        likedArtists.add(card.getTrack().getArtists().get(0));
+        final Queue<TrackFull>[] likedTracksQ = new Queue[]{new LinkedList<TrackFull>()};
+        final Queue<ArtistSimplified>[] likedArtistsQ = new Queue[]{new LinkedList<ArtistSimplified>()};
+        final Queue<String>[] likedGenresQ = new Queue[]{new LinkedList<String>()};
+
+
+        ParseQuery<ParseObject> query = ParseQuery.getQuery("LikedObjects");
+        query.whereEqualTo("user", currentUser.getObjectId());
+        query.findInBackground(new FindCallback<ParseObject>() {
+            @Override
+            public void done(List<ParseObject> objects, ParseException e) {
+                if (e == null) {
+                    // No exception, so we have successfully found corresponding LikedObjects for user.
+                    // Note that there should only be one row corresponding to each user, so we only
+                    // need to retrieve the first out of this returned list.
+                    ParseObject obj = objects.get(0);
+
+                    likedTracksQ[0] = (LinkedList<TrackFull>) currentUser.get("likedTracks");
+                    likedArtistsQ[0] = (LinkedList<ArtistSimplified>) currentUser.get("likedArtists");
+                    likedGenresQ[0] = (LinkedList<String>) currentUser.get("likedGenres");
+                } else {
+                    Log.e(TAG, "error in query: " + e.getStackTrace());
```

Comment:
a useful log may be: `did not find corresponding LikedObjects for the user: + <user's name>`